### PR TITLE
Make it so that `clinic` is never `None`

### PR DESCRIPTION
**Pull request by AlexWaygood**

### CODE REVIEW

**Comment by erlend-aasland** [comment]
I'll review this tomorrow

File: Lib/test/test_clinic.py [hunk 1]
```diff
@@ -812,9 +791,12 @@ def test_clinic_1(self):
 
 
 class ClinicParserTest(TestCase):
+    def tearDown(self):
+        if hasattr(clinic, "clinic"):
+            del clinic.clinic
 
     def parse(self, text):
-        c = _make_clinic()
+        c = clinic.clinic = _make_clinic()
```
Comment (erlend-aasland) [inline Lib/test/test_clinic.py:799]: Why are we setting `clinic` in `clinic`? That also seems like a hack we'd like to get rid of.

Comment (AlexWaygood) [reply Lib/test/test_clinic.py:799]: I agree that it's ugly.

The issue is that several tests in this TestCase subclass currently implicitly rely on the fact that whenever a `Clinic` object is created, this also assigns a module-level `clinic` variable for the `clinic` module, due to this hack:

https://github.com/Argument-Clinic/cpython/blob/73507382ac184a72b59ebb0c2f85e8b1d2dfa58e/Tools/clinic/clinic.py#L2338-L2339

In order to remove this hack, we have to make it explicit in the test suite exactly which tests depend on there being a module-level `clinic` variable on the `clinic` module.

File: Tools/clinic/clinic.py [hunk 2]
```diff
@@ -2458,23 +2450,28 @@ def __repr__(self) -> str:
         return "<clinic.Clinic object>"
 
 
+clinic: Clinic
+
+
 def parse_file(
         filename: str,
         *,
+        parser: argparse.ArgumentParser,
```
Comment (erlend-aasland) [inline Tools/clinic/clinic.py:2459]: Perhaps a positional argument?

Comment (erlend-aasland) [reply Tools/clinic/clinic.py:2459]: (not blocking this PR, though)

**Review by erlend-aasland — APPROVED** [review]

File: Tools/clinic/clinic.py (hunk 2 above)
Comment (AlexWaygood) [reply Tools/clinic/clinic.py:2459]: I've no strong opinions either way :) I'll make it a positional-or-keyword arg and take it over to CPython

**Comment by erlend-aasland** [comment]
Did we push this upstream?

**Comment by AlexWaygood** [comment]
> Did we push this upstream?

We didn't. I wasn't sure if it was really a good idea or not, because it didn't tackle the underlying issue, of which this issue is a symptom (it doesn't get rid of the global `clinic` variable!).

I'm still not sure about this change, so I'll reject my own idea and close this :)

**Pull request closed without merging**